store/cachemulti/store.go

```diff
@@ -23,8 +24,9 @@ type Store struct {
 	stores map[types.StoreKey]types.CacheWrap
 	keys   map[string]types.StoreKey
 
-	traceWriter  io.Writer
-	traceContext types.TraceContext
+	traceWriter       io.Writer
+	traceContext      types.TraceContext
+	traceContextMutex sync.RWMutex
```

Comment:
should the mutex be a pointer?

Reply:
Just rebased on top of 0.45.5 instead, and that seems to have fixed the issue. Thanks Evan! Appreciate your help :)